src/main.rs

```diff
@@ -0,0 +1,77 @@
+mod dockerfile;
+
+use dockerfile::{Dockerfile, ValidationResult};
+use glob::glob;
+use std::{error::Error, process};
+
+fn main() {
```

Comment:
`main` is allowed to return a result, which might allow you to remove the `pass`/`skip`/`fail` methods.  See https://rust-cli.github.io/book/tutorial/errors.html for more on that.

Reply:
yeah i saw that, but thought i'd make the exit code explicit here since that's what github actions uses to determine success / failure

Reply:
Not really sure to be honest - haven't worked with this enough! Maybe worth trying both versions and seeing which you prefer?